sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -604,7 +604,11 @@ case class ToUnixTimestamp(
        1460041200
   """,
   since = "1.5.0")
-case class UnixTimestamp(timeExp: Expression, format: Expression, timeZoneId: Option[String] = None)
+case class UnixTimestamp(
+    timeExp: Expression,
+    format: Expression,
+    timeZoneId: Option[String] = None,
+    override val downScaleFactor: Long = MICROS_PER_SECOND)
```

Comment:
ditto, this violates the semantic of `UnixTimestamp`, which promises to return seconds.

We should create a new expression. We can create a base class for `UnixTimestamp` and the new expression to reuse code.